### PR TITLE
Add the joist class.

### PR DESCRIPTION
**Pull request by ikeough**
BACKGROUND:
Logistical buildings, warehouses, and industrial facilities often have roofs which are made of a combination of joist girders and bar joists from companies like Vulkraft. The Elements library does not currently contain element types which can represent these structures.

DESCRIPTION:
- This PR adds the `Joist` class. The `Joist` class is capable of representing both bar joists and joist girders.

TESTING:
- See `StructuralFramingTests.Joists`
  
FUTURE WORK:
- This is a first pass on joists. Currently we represent all webs using LL sections. Representing bar joists will require representing webs as bars.

REQUIRED:
- [ ] All changes are up to date in `CHANGELOG.md`. **n/a**

image

---
This change is [Reviewable](https://reviewable.io/reviews/hypar-io/elements/691)

### CODE REVIEW

**Review by katehryhorenko — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hypar-io/elements/691)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @ikeough)

___
*[Elements/src/Joist.cs, line 67 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXWlrI9plm-YplrGyE:-MnXWlrI9plm-YplrGyF:b95bdm) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L67)):*
> ```Smalltalk
>         IN72
>     }
>     /// <summary>
> ```

please, add line here
___
*[Elements/src/Joist.cs, line 139 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXX1DM4FAjD9he-YV7:-MnXX1DM4FAjD9he-YV8:bbrvmxx) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L139)):*
> ```Smalltalk
>                      Guid id = default) : base(curve, null, material, name: name, id: id)
>         {
>             if (!(curve is Line))
> ```

Maybe it makes sense to use Line as input parameter instead?
___
*[Elements/src/Joist.cs, line 155 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXZfr40rdCCX5BWUCX:-MnXZfr40rdCCX5BWUCY:b-fmxlx5) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L155)):*
> ```Smalltalk
>         }
> 
>         private Profile[] LL(JoistProfileType profileType, bool flip = false)
> ```

Can we rename this method? 
var ll = LL(profileType, flip) looks like we are calling constructor :slightly_smiling_face: 
___
*[Elements/src/Joist.cs, line 159 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXYxU4D_yYzlm3IkJZ:-MnXYxU4D_yYzlm3IkJ_:bsxpyx3) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L159)):*
> ```Smalltalk
>             var w = Units.InchesToMeters(double.Parse(profileType.ToString().Substring(2).Split('X')[0].Replace('_', '.')));
> 
>             var L = Polygon.L(w, w, Units.InchesToMeters(0.125));
> ```

Can we hardcode somewhere 0.125 constant? 
___
*[Elements/src/Joist.cs, line 162 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXXtE3E2m_h9N7Ehvw:-MnXXtE3E2m_h9N7Ehvx:b-3x1r05) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L162)):*
> ```Smalltalk
>             double flangeT = Units.InchesToMeters(0.125);
> 
>             Transform r;
> ```

It's always a good idea to use more meaningful names, to simplify code reading. It's a little bit hard to follow the code when it has one latter names
___
*[Elements/src/Joist.cs, line 303 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXZUsj5BairJDNNeAi:-MnXZUsj5BairJDNNeAj:b-c59hrr) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L303)):*
> ```Smalltalk
>                 else
>                 {
> ```

extra line

**Review by ikeough — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hypar-io/elements/691)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @katehryhorenko)

___
*[Elements/src/Joist.cs, line 67 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXWlrI9plm-YplrGyE:-Mnc02KvCKDaJIG4QgEF:b-896fix) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L67)):*
<details><summary><i>Previously, katehryhorenko (Kateryna Hryhorenko) wrote…</i></summary><blockquote>

please, add line here
</blockquote></details>

Done.
___
*[Elements/src/Joist.cs, line 139 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXX1DM4FAjD9he-YV7:-Mnc0DCEEkn1ebkJ_Ts5:b-a3fl9f) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L139)):*
<details><summary><i>Previously, katehryhorenko (Kateryna Hryhorenko) wrote…</i></summary><blockquote>

Maybe it makes sense to use Line as input parameter instead?
</blockquote></details>

Haha. Yes, of course this makes more sense :)
___
*[Elements/src/Joist.cs, line 155 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXZfr40rdCCX5BWUCX:-Mnc0S-tAkBBUUlc8Hb0:b-896fix) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L155)):*
<details><summary><i>Previously, katehryhorenko (Kateryna Hryhorenko) wrote…</i></summary><blockquote>

Can we rename this method? 
var ll = LL(profileType, flip) looks like we are calling constructor :slightly_smiling_face: 
</blockquote></details>

Done.
___
*[Elements/src/Joist.cs, line 159 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXYxU4D_yYzlm3IkJZ:-Mnc0kZcCx3kHku3OjBU:b-896fix) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L159)):*
<details><summary><i>Previously, katehryhorenko (Kateryna Hryhorenko) wrote…</i></summary><blockquote>

Can we hardcode somewhere 0.125 constant? 
</blockquote></details>

Done.
___
*[Elements/src/Joist.cs, line 162 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXXtE3E2m_h9N7Ehvw:-Mnc11liBNGpLoQhgoJY:b-896fix) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L162)):*
<details><summary><i>Previously, katehryhorenko (Kateryna Hryhorenko) wrote…</i></summary><blockquote>

It's always a good idea to use more meaningful names, to simplify code reading. It's a little bit hard to follow the code when it has one latter names
</blockquote></details>

Done.
___
*[Elements/src/Joist.cs, line 303 at r1](https://reviewable.io/reviews/hypar-io/elements/691#-MnXZUsj5BairJDNNeAi:-Mnc16uI14wQwz1OL954:b-896fix) ([raw file](https://github.com/hypar-io/elements/blob/f06a796befe1fe33907a3e5e72e6e92862e02956/Elements/src/Joist.cs#L303)):*
<details><summary><i>Previously, katehryhorenko (Kateryna Hryhorenko) wrote…</i></summary><blockquote>

extra line
</blockquote></details>

Done.
